app/src/main/java/io/gnosis/safe/ui/StartActivity.kt

```diff
@@ -164,8 +166,27 @@ class StartActivity : BaseActivity(), SafeOverviewNavigationHandler {
         }
     }
 
+    private fun adjustSafeNameWidth() {
+        with(toolbarBinding) {
+            val bounds = Rect()
+            safeName.paint.getTextBounds(safeName.text.toString(), 0, safeName.text.length, bounds)
+            val safeNameLength = bounds.right - bounds.left
+
+            // wait till views are measured
+            safeName.post {
+                val readOnlySpace = readOnly.measuredWidth + readOnly.marginLeft + readOnly.marginRight
+                if (safeNameLength > safeAddress.measuredWidth - readOnlySpace) {
+                    safeName.width = safeNameLength - (safeName.marginLeft + safeName.marginRight) - readOnlySpace
```

Comment:
There might be something wrong with the calculation:
![image](https://user-images.githubusercontent.com/246473/115019948-28c46f80-9eba-11eb-935d-ad38626e230b.png)